### ISSUE TITLE
Nested model attribute label

### ISSUE THREAD

**Issue by tekord** [yiisoft/yii2#14360]
### What steps will reproduce the problem?

Define a class with nested models:

```php
class CompletedOrderReviewForm extends Model {
   /** @var OrderReview */
   public $summaryReview;

   /** @var OrderReview */
   public $primaryReview;

   /** @var OrderReview[] */
   public $otherReviews = [];

	public function rules() {
		return [
			[['summaryReview', 'primaryReview', 'secondaryReviews'], 'safe'],
                        // some complex validation here
		];
	}
}
```
Render inputs through ActiveForm:

```php
<?= $form->field($formModel, 'summaryReview[rate]')
		->textInput([
			'type' => 'number'
		]) ?>
```

### What is the expected result?

The expected label of the input is 'Rate'.

### What do you get instead?

The label of this input is 'Summary Review'.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12
| PHP version      | 7.1.4
| Operating system | Windows 8.1 64 bit

**Comment by dynasource** [comment]
within the current state of the software, this is expected behavior.

In order to get the right label, you should use:

```php
 $form->field($formModel->summaryReview, 'rate')
		->textInput([
			'type' => 'number'
		]) ?>
```

**Comment by TerraSkye** [comment]
@dynasource woulnt the field name be incorrect then?

**Comment by dynasource** [comment]
correct. To my knowledge, this is could be further improved.

**Comment by tekord** [comment]
`$form->field($formModel->summaryReview, 'rate')` is another model, which I have to load and validate individually. I want this model to be nested for convenience.

**Comment by SamMousa** [comment]
@TerraSkye AFAIK you could override `getAttributeLabel` and implement the label resolution there. Ideally you forward it so it takes the label from the related model.

**Comment by tekord** [comment]
@SamMousa overridding of `getAttributeLabel` is not working. My form has inputs with names `CompletedOrderReviewForm[**primaryReview**][rate]`, `CompletedOrderReviewForm[**summaryReview**][rate]` and `CompletedOrderReviewForm[**secondaryReviews**][rate]`, but `CompletedOrderReviewForm::getAttribute` receives only `primaryReview`, `summaryReview` and `secondaryReviews` as arguments. There is not way to know what field's label of my nested model requested.

**Comment by SamMousa** [comment]
Ah ok;don't know how to get around that.

**Comment by TerraSkye** [comment]
the only workaround i see is to manually set the label.

**Comment by dynasource** [comment]
I agree this to be an issue. Hasnt this been posted before in another ticket?

**Comment by tekord** [comment]
@TerraSkye do you mean $form->field(...)->label(...)? Field name would be ok, but messages in validation errors are not.

**Comment by TerraSkye** [comment]
personally i had bad experience by nesting models, the way u presented and therefore i tend to avoid that situation, but yes i understand the issue.
